src/components/Account.jsx

```
@@ -0,0 +1,218 @@
+            alert("비밀번호가 일치하지 않습니다.");
+            break;
+          default:
+            break;
+        }
+        break;
+      case "password2":
+        switch (true) {
+          case confirmPassword === "":
+            break;
+          case !isValidPassword(confirmPassword):
+            alert("비밀번호는 영문, 숫자 조합 8자 이상을 입력해 주세요.");
+            break;
+          case password !== "" && password !== confirmPassword:
+            alert("비밀번호가 일치하지 않습니다.");
+            break;
+          default:
+            break;
+        }
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (isSignin) {
+      if (email === "test@codeit.com" && password === "codeit101") {
+        navigate("/my-link");
+      } else {
+        alert("이메일과 비밀번호를 확인해주세요.");
+      }
+    } else {
+      switch (true) {
+        case email === "":
+          alert("이메일을 입력해 주세요.");
+          break;
+        case !isValidEmail(email):
+          alert("올바른 이메일 형식이 아닙니다.");
+          break;
+        case email === "test@codeit.com":
+          alert("이미 사용 중인 이메일입니다.");
+          break;
+        case password === "" || confirmPassword === "":
+          alert("비밀번호를 입력해 주세요.");
+          break;
+        case !isValidPassword(password) || !isValidPassword(confirmPassword):
+          alert("비밀번호는 영문, 숫자 조합 8자 이상을 입력해 주세요.");
+          break;
+        case password !== confirmPassword:
+          alert("비밀번호가 일치하지 않습니다.");
+          break;
+        default:
+          navigate("/my-link");
+          break;
+      }
+    }
+  };
+
+  const [showPassword, setShowPassword] = useState(false);
+  const [showPassword2, setShowPassword2] = useState(false);
+
+  const handlePasswordToggler = (e) => {
+    e.preventDefault();
+    setShowPassword((prevState) => !prevState);
+  };
+
+  const handlePassword2Toggler = (e) => {
+    e.preventDefault();
+    setShowPassword2((prevState) => !prevState);
+  };
```

혹시 여기 preventDefault 안하면 어떻게 되나요? form submit 이 되진 않을거 같은데 걸어두신 이유가 있나요?

---

preventDefault를 안하면 password hide/show 기능을 하는 eye toggler icon을 클릭했을 때, 만약 focus 되었던 input tag가 있다면 blur 처리 되면서 의도하지 않게 alert box가 뜹니다.  이런 UX는 좋지 않다고 생각하여 eye toggler icon을 클릭해도 focus 된 상태의 input tag의 focus 상태를 유지하기 위해 preventDefault를 사용했습니다.